### PR TITLE
Add ignoreUnits option to number-max-precision rule

### PR DESCRIPTION
**Pull request by sendilkumarn**
> Which issue, if any, is this issue related to?

#2785 

> Is there anything in the PR that needs further explanation?

No, it's self explanatory

### CODE REVIEW

File: lib/rules/number-max-precision/README.md [hunk 1]
```diff
@@ -37,3 +37,46 @@ a { top: 3.24px; }
 ```css
 @media (min-width: 3.23em) {}
 ```
+## Optional secondary options
```
Comment (jeddy3) [inline lib/rules/number-max-precision/README.md]: Please add one blank line above the heading.

(Note: Let's see if remark can lint for this).

File: lib/rules/number-max-precision/README.md [hunk 2]
```diff
@@ -37,3 +37,46 @@ a { top: 3.24px; }
...
+
+### `ignoreUnits: ["string"]`
```
Comment (jeddy3) [inline lib/rules/number-max-precision/README.md]: The `optionsMatches` util accepts regular expressions as well as strings. Can you update the examples to reflect this please, [like so](https://github.com/stylelint/stylelint/tree/master/lib/rules/unit-no-unknown#optional-secondary-options)? e.g. 

```js
["/^my-/", "%"]
```

```css
a { top: 3.245my-unit; }
```

File: lib/rules/number-max-precision/__tests__/index.js [hunk 3]
```diff
@@ -150,3 +150,50 @@ testRule(rule, {
     }
   ]
 });
+
+testRule(rule, {
+  ruleName,
+  config: [0, { ignoreUnits: ["%"] }],
```
Comment (jeddy3) [inline lib/rules/number-max-precision/__tests__/index.js]: Can we get update the config to:

```
ignoreUnits: ["%", "/^my-/"]
```

And get an accept and reject test for the `"/^my-/"` value.

**Comment by sendilkumarn** [comment]
@jeddy3 done the changes. 👍

File: lib/rules/number-max-precision/README.md [hunk 4]
```diff
@@ -37,3 +37,57 @@ a { top: 3.24px; }
 ```css
 @media (min-width: 3.23em) {}
 ```
+
+## Optional secondary options
+
+### `ignoreUnits: ["string"]`
```
Comment (jeddy3) [inline lib/rules/number-max-precision/README.md]: `### ignoreUnits: ["/regex/", "string"]`

File: lib/rules/number-max-precision/README.md [hunk 5]
```diff
@@ -37,3 +37,57 @@ a { top: 3.24px; }
...
+
+Ignore number-max-precision rule for the value with the unit.
```
Comment (jeddy3) [inline lib/rules/number-max-precision/README.md]: Can we use the following description, please:

"Ignore the precision of numbers for values with the specified units."

As it's more consistent with the descriptions [else where](https://github.com/stylelint/stylelint/search?l=Markdown&p=1&q=ignore&type=&utf8=%E2%9C%93).

**Comment by jeddy3** [comment]
> -   Added: `ignoreUnits` option to `number-max-precision` ([#2941](https://github.com/stylelint/stylelint/pull/2941)).

**Pull request merged**